python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -336,6 +335,45 @@ def _set_state(self, state: TrainControllerState):
         for callback in self._controller_callbacks:
             callback.after_controller_state_update(previous_state, state)
 
+    def _make_and_handle_scaling_decision_for_non_running_worker_group(
+        self,
+        controller_state: TrainControllerState,
+        next_state_cls: Type[TrainControllerState],
+    ) -> TrainControllerLoopIterationResult:
+        """Make a scaling decision for a non-running worker group and return the appropriate next state.
+
+        This method should be called when entering a state that requires a scaling decision
+        for a non-running worker group.
+
+        This method handles the complete flow of:
+        1. Getting a scaling decision for a non-running worker group
+        2. Determining the next state based on the decision type
+        3. Creating and returning the iteration result
+
+        Args:
+            controller_state: The current controller state
+            next_state_cls: The state class to use when a NoopDecision is made.
+
+        Returns:
+            TrainControllerLoopIterationResult with the appropriate next state
+        """
+        scaling_decision = (
+            self._scaling_policy.make_decision_for_non_running_worker_group()
+        )
+
+        if isinstance(scaling_decision, NoopDecision):
+            next_state = next_state_cls()
+        elif isinstance(scaling_decision, ResizeDecision):
+            next_state = SchedulingState(scaling_decision)
+        else:
+            raise ValueError(f"Unexpected scaling decision: {scaling_decision}")
+
+        return TrainControllerLoopIterationResult(
+            run_attempt_id=self._get_run_attempt_id(),
+            previous_state=controller_state,
+            next_state=next_state,
```

Comment:
remove the `next_state_cls` and just set next_state to `controller_state` if noop?